frontend/dockerfile/instructions/bflag.go

```diff
@@ -180,27 +173,27 @@ func (bf *BFlags) Parse() error {
 		switch flag.flagType {
 		case boolType:
 			// value == "" is only ok if no "=" was specified
-			if index >= 0 && value == "" {
+			if hasValue && value == "" {
 				return errors.Errorf("missing a value on flag: %s", arg)
 			}
 
-			lower := strings.ToLower(value)
-			if lower == "" {
+			switch strings.ToLower(value) {
+			case "true", "":
 				flag.Value = "true"
-			} else if lower == "true" || lower == "false" {
-				flag.Value = lower
-			} else {
+			case "false":
+				flag.Value = "false"
```

Comment:
Not sure why we've not used `strconv.ParseBool` here? We should probably use that instead, to be consistent with everywhere else where we do that.

Reply:
Possibly because we wanted to define our own API, and `strconv.ParseBool` is quite relaxed ("t, T, f, F, 1, 0, true, false, TRUE, FALSE, True, False" are all accepted), which can become confusing quickly; do you know if making it more relaxed would affect caching? (e.g. changing `foo=true` to `foo=TRUE` now causing a cache-miss)?

Reply:
It shouldn't affect caching, but agreed, it is an API change, and I'm not sure the alternative syntaxes are any easier to read/write - let's leave as is for now.